### PR TITLE
Add OVHcloud Getting Started

### PR DESCRIPTION
**Pull request by raclepoulpe**
This is the OVHcloud guide to install Fury 1.24 on OVHcloud Managed Kubernetes.

### CODE REVIEW

File: fury-on-ovhcloud/infrastructure/terraform/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# COMING SOON
```
Comment (ralgozino) [inline fury-on-ovhcloud/infrastructure/terraform/README.md:1]: You can use [`terraform-docs`](https://terraform-docs.io/) to generate a basic readme describing the terraform module 😉 

You can see a sample configuration file that we use here:
https://github.com/sighupio/fury-eks-installer/blob/master/modules/eks/.terraform-docs.yml

File: fury-on-ovhcloud/README.md [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+```
+
+Wait a few minutes until the end of the deployement.
+
+### Configure your kubectl environment
+
+Once the Managed Kubernetes Cluster created, in the `terraform` folder, you get the associated `kubeconfig` file. Set the KUBECONFIG variable value like this:
+
+```bash
+export KUBECONFIG="$(pwd)/kubeconfig"
```
Comment (ralgozino) [inline fury-on-ovhcloud/README.md:357]: better to use $PWD (an env var), than catching the output of a subshell with `$()`:
```suggestion
export KUBECONFIG="$PWD/kubeconfig"
```

File: fury-on-ovhcloud/README.md [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+- `forecastle.fury.info`
+- `grafana.fury.info`
+- `opensearch-dashboards.fury.info`
+
+To access the ingresses more easily via the browser, configure your local DNS to resolve the ingresses to the external load balancer IP:
+
+1. Get the address of the external load balancer:
+
+```bash
+kubectl get svc -n ingress-nginx ingress-nginx --no-headers | awk '{print $4}'
```
Comment (ralgozino) [inline fury-on-ovhcloud/README.md:435]: I NEED TO TEST THIS, but I'm 99% sure it works

No need to pipe to `awk`, you can tell kubectl what to print out using the `-o jsonpath=<template>` flag:
```suggestion
kubectl get svc -n ingress-nginx ingress-nginx -ojsonpath='{.spec.externalIPs[*]}'
```

File: fury-on-ovhcloud/README.md [hunk 4]
```diff
@@ -0,0 +1,539 @@
...
+```bash
+cd ./terraform/
+terraform destroy -var-file=variables.tfvars
+```
+
+## Conclusions
+
+Congratulations, you made it! 🥳🥳
+
+We hope you enjoyed this tour of Fury!
```
Comment (ralgozino) [inline fury-on-ovhcloud/README.md:504]:
```suggestion
We hope you enjoyed this tour of Fury on OVHcloud!
```

File: fury-on-ovhcloud/README.md [hunk 5]
```diff
@@ -0,0 +1,539 @@
...
+
+| Terraform Provider | Credentials |
+|---|---|
+| [OVH Provider](https://registry.terraform.io/providers/ovh/ovh/latest/docs) | [OVHcloud API](https://docs.ovh.com/gb/en/api/first-steps-with-ovh-api/) credentials |
+| [Openstack Provider](https://registry.terraform.io/providers/terraform-provider-openstack/openstack/latest/docs) | [Openstack user](https://docs.ovh.com/sg/en/public-cloud/creation-and-deletion-of-openstack-user/) credentials |
+
+### Pre-requisites
+
+The tools we need are `furyctl`, `terraform`, `kubectl` and `kustomize`.
+
```
Comment (ralgozino) [inline fury-on-ovhcloud/README.md:41]: We should mention that you can also use the container that we make, like in the other guides.

And that there's a script provided to run all the installation commands at once.

File: fury-on-ovhcloud/README.md [hunk 6]
```diff
@@ -0,0 +1,539 @@
...
+Install the latest `furyctl` cli from the [Github Furyctl Realise page](https://github.com/sighupio/furyctl/releases).
+
+Example on an Ubuntu distribution:
+
+```bash
+wget -q "https://github.com/sighupio/furyctl/releases/download/v0.9.0/furyctl-$(uname -s)-amd64" -O /tmp/furyctl \
+&& chmod +x /tmp/furyctl \
+&& sudo mv /tmp/furyctl /usr/local/bin/furyctl
+```
+
```
Comment (ralgozino) [inline fury-on-ovhcloud/README.md:55]:
```suggestion

See [furyctl's readme](https://github.com/sighupio/furyctl) for more installation methods.

```

**Review by ralgozino — CHANGES_REQUESTED** [review]
Hi Olivier! thank you so much for the PR.

I'm suggesting a first round of changes in style, typos, etc.

We'll do a second round when we test the guide most probably.

Let me know if you need anything from our side

File: fury-on-ovhcloud/README.md [hunk 7]
```diff
@@ -0,0 +1,539 @@
...
+- **OVHcloud Openstack User** - To manage the network environment with the Terraform provider, you must have an Openstack user. Use this [guide](https://docs.ovh.com/gb/en/public-cloud/creation-and-deletion-of-openstack-user/) to create one.
+
+## Step 1 - Automatic provisioning of a Managed Kubernetes Cluster in a private network
+
+We are using the `terraform` cli to automaticly deploy the private network, and then the Managed Kubernetes Cluster.
+
+| Terraform Provider | Credentials |
+|---|---|
+| [OVH Provider](https://registry.terraform.io/providers/ovh/ovh/latest/docs) | [OVHcloud API](https://docs.ovh.com/gb/en/api/first-steps-with-ovh-api/) credentials |
+| [Openstack Provider](https://registry.terraform.io/providers/terraform-provider-openstack/openstack/latest/docs) | [Openstack user](https://docs.ovh.com/sg/en/public-cloud/creation-and-deletion-of-openstack-user/) credentials |
```
[the remark this answers is not in the document]
Comment (raclepoulpe) [reply fury-on-ovhcloud/README.md:36]: I don't see the meaning of this one :)

File: fury-on-ovhcloud/infrastructure/terraform/README.md (hunk 1 above)
Comment (raclepoulpe) [reply fury-on-ovhcloud/infrastructure/terraform/README.md]: Thx for the tip!
I used it for the main terraform script and the modules taht are used

File: fury-on-ovhcloud/README.md [hunk 8]
```diff
@@ -0,0 +1,539 @@
...
+
+## Prerequisites
+
+This tutorial assumes some basic familiarity with Kubernetes and [OVHcloud](https://www.ovhcloud.com/). Some experience with Terraform is helpful but not required.
+
+To follow this tutorial, you need:
+
+- **OVHcloud Account** - You must have an active account to use OVHcloud services. Use this [guide](https://docs.ovh.com/gb/en/customer/create-ovhcloud-account/) to create one.
+- **OVHcloud Public Cloud project** - You must have a Public Cloud Project. Use this [guide](https://docs.ovh.com/gb/en/public-cloud/create_a_public_cloud_project/) to create one.
+- **OVHcloud Openstack User** - To manage the network environment with the Terraform provider, you must have an Openstack user. Use this [guide](https://docs.ovh.com/gb/en/public-cloud/creation-and-deletion-of-openstack-user/) to create one.
```
[the remark this answers is not in the document]
Comment (raclepoulpe) [reply fury-on-ovhcloud/README.md]: Done (thx)

File: fury-on-ovhcloud/README.md [hunk 9]
```diff
@@ -0,0 +1,539 @@
...
+
+## Step 1 - Automatic provisioning of a Managed Kubernetes Cluster in a private network
+
+We are using the `terraform` cli to automaticly deploy the private network, and then the Managed Kubernetes Cluster.
```
[the remark this answers is not in the document]
Comment (raclepoulpe) [reply fury-on-ovhcloud/README.md]: Done (thx)

File: fury-on-ovhcloud/README.md [hunk 10]
```diff
@@ -0,0 +1,539 @@
...
+
+### Pre-requisites
+
+The tools we need are `furyctl`, `terraform`, `kubectl` and `kustomize`.
+
+Click on the desired tool to see how to install it:
+
+<details><summary>1 - furyctl</summary>
+
+Install the latest `furyctl` cli from the [Github Furyctl Realise page](https://github.com/sighupio/furyctl/releases).
```
[the remark this answers is not in the document]
Comment (raclepoulpe) [reply fury-on-ovhcloud/README.md]: Done (thx)

File: fury-on-ovhcloud/README.md [hunk 11]
```diff
@@ -0,0 +1,539 @@
...
+
+```bash
+wget -q "https://github.com/sighupio/furyctl/releases/download/v0.9.0/furyctl-$(uname -s)-amd64" -O /tmp/furyctl \
+&& chmod +x /tmp/furyctl \
+&& sudo mv /tmp/furyctl /usr/local/bin/furyctl
+```
+
+</details>
+
+<details><summary>2 - terraform cli</summary>
```
[the remark this answers is not in the document]
Comment (raclepoulpe) [reply fury-on-ovhcloud/README.md]: Done (thx)

File: fury-on-ovhcloud/README.md [hunk 12]
```diff
@@ -0,0 +1,539 @@
...
+
+Example:
+
+```bash
+export OVH_CLOUD_PROJECT_SERVICE="xxxxxxxxxxxxxxxxxxxxx"
+```
+
+That's all you need to use the OVHcloud Terraform Provider.
+
+#### 2 - OVHcloud Openstack User variables
```
[the remark this answers is not in the document]
Comment (raclepoulpe) [reply fury-on-ovhcloud/README.md]: Done (thx)

File: fury-on-ovhcloud/README.md [hunk 13]
```diff
@@ -0,0 +1,539 @@
...
+
+[Get your Openstack user's `openrc` file](https://docs.ovh.com/gb/en/public-cloud/set-openstack-environment-variables/#step-1-retrieve-the-variables) to extract and to set necessary variables:
```
[the remark this answers is not in the document]
Comment (raclepoulpe) [reply fury-on-ovhcloud/README.md]: Done (thx)

File: fury-on-ovhcloud/README.md [hunk 14]
```diff
@@ -0,0 +1,539 @@
...
+export OS_USER_DOMAIN_NAME=${OS_USER_DOMAIN_NAME:-"Default"}
+export OS_PROJECT_DOMAIN_NAME=${OS_PROJECT_DOMAIN_NAME:-"Default"}
+export OS_TENANT_ID="xxxxxxxxxxxxxxxxxx"
+export OS_TENANT_NAME="xxxxxxxxxxxxxxxx"
+export OS_USERNAME="user-xxxxxxxxxxxxxx"
+export OS_PASSWORD="xxxxxxxxxxxxxxxxxxx"
+export OS_REGION_NAME="xxx"
+```
+
+You are ready to use the Openstack Terraform Provider.
```
[the remark this answers is not in the document]
Comment (raclepoulpe) [reply fury-on-ovhcloud/README.md]: Done (thx)

File: fury-on-ovhcloud/README.md [hunk 15]
```diff
@@ -0,0 +1,539 @@
...
+terraform init
+terraform plan -var-file=variables.tfvars
+terraform apply -var-file=variables.tfvars
+```
+
+Wait a few minutes until the end of the deployement.
+
+### Configure your kubectl environment
+
+Once the Managed Kubernetes Cluster created, in the `terraform` folder, you get the associated `kubeconfig` file. Set the KUBECONFIG variable value like this:
```
[the remark this answers is not in the document]
Comment (raclepoulpe) [reply fury-on-ovhcloud/README.md]: Done (thx)

File: fury-on-ovhcloud/README.md (hunk 2 above)
Comment (raclepoulpe) [reply fury-on-ovhcloud/README.md]: Done (thx)

File: fury-on-ovhcloud/README.md [hunk 16]
```diff
@@ -0,0 +1,551 @@
...
+- patches the upstream modules (e.g. `patches/opensearch-resources.yml` limits the resources requested by OpenSearch)
+- deploys some additional custom resources (e.g. `resources/ingress.yml`)
+
+Install the modules:
+
+```bash
+cd manifests/
+
+make apply
+# Due to some chicken-egg 🐓🥚 problem with custom resources you have to apply multiple times
```
Comment (ralgozino) [inline fury-on-ovhcloud/README.md:426]:
```suggestion
# Due to some CRDs being created, the first time you have to run make apply multiple times. Run it until you see no more errors.
```

File: fury-on-ovhcloud/infrastructure/terraform/variables.tf [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+  })
+}
+
+# Managed Kubernetes Cluster
+
+variable "kube" {
+  description = "Managed Kubernetes Cluster parameters"
+  type = object({
+    name            = string
+    pv_network_name = string
```
Comment (ralgozino) [inline fury-on-ovhcloud/infrastructure/terraform/variables.tf:44]:
```suggestion
    version         = optional(string)
    pv_network_name = string
```

File: fury-on-ovhcloud/infrastructure/terraform/variables.tfvars [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+# Network - Router
+
+router = {
+  name = "furyRouter"
+}
+
+# Managed Kubernetes Cluster
+
+kube = {
+  name            = "furykubernetesCluster"
```
Comment (ralgozino) [inline fury-on-ovhcloud/infrastructure/terraform/variables.tfvars:29]:
```suggestion
  version         = "1.24"
```

File: fury-on-ovhcloud/infrastructure/terraform/modules/kubernetes/k8s_pvnw/main.tf [hunk 19]
```diff
@@ -0,0 +1,20 @@
+resource "ovh_cloud_project_kube" "kube" {
+  name   = var.kube.name
```
Comment (ralgozino) [inline fury-on-ovhcloud/infrastructure/terraform/modules/kubernetes/k8s_pvnw/main.tf:2]:
```suggestion
  version = var.kube.version
```

File: fury-on-ovhcloud/infrastructure/terraform/modules/kubernetes/k8s_pvnw/variables.tf [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+  description = "Region"
+  type        = string
+}
+
+# Managed Kubernetes Cluster
+
+variable "kube" {
+  description = "Managed Kubernetes Cluster parameters"
+  type = object({
+    name            = string
```
Comment (ralgozino) [inline fury-on-ovhcloud/infrastructure/terraform/modules/kubernetes/k8s_pvnw/variables.tf:13]:
```suggestion
    version         = optional(string)
```

**Review by ralgozino — CHANGES_REQUESTED** [review]
we are almost there! some minor corrections and improvemets suggestions

**Review by ralgozino — CHANGES_REQUESTED** [review]
as per previous comment and conversations

**Review by ralgozino — APPROVED** [review]
LGTM 🎉

**Comment by raclepoulpe** [comment]
Yeah ! 👍💥

Le mer. 22 févr. 2023 à 12:10, Ramiro Algozino ***@***.***> a
écrit :

**Pull request merged**